### PR TITLE
fix: page templates date field required

### PR DESCRIPTION
**Pull request by priscila-moneo**
ref: https://app.clickup.com/t/9014802374/86b9c2nd4

## Summary by CodeRabbit

* **New Features**
  * Global sponsor page templates now adapt the media request form by conditionally showing the upload deadline.
* **Bug Fixes**
  * Upload deadline validation now correctly differs between global and non-global templates.
  * Stale/legacy `upload_deadline` values are cleared and omitted when the field is hidden.
  * Save/close interactions are safer while saving, and successful saves no longer automatically dismiss popups across pages.
  * Failed saves no longer trigger list refresh behavior.
* **Tests**
  * Added/updated Jest coverage for global vs. non-global rendering, validation, normalization, and save-state guarding.

### CODE REVIEW

File: src/pages/sponsors-global/page-templates/page-template-popup/page-template-modules-form.js [hunk 1]
```diff
@@ -129,7 +129,13 @@ const PageModules = ({ name = "modules", getAllMediaFileTypes }) => {
       case PAGES_MODULE_KINDS.DOCUMENT:
         return <DocumentDownloadModule baseName={name} index={index} />;
       case PAGES_MODULE_KINDS.MEDIA:
-        return <MediaRequestModule baseName={name} index={index} />;
+        return (
+          <MediaRequestModule
+            baseName={name}
+            index={index}
+            isGlobal={isGlobal}
```
Comment (santipalenque) [inline src/pages/sponsors-global/page-templates/page-template-popup/page-template-modules-form.js]: at this point isGlobal means nothing, can we change it to showUploadDeadline={!isGlobal} instead ? and add a default to isGlobal

File: src/pages/sponsors-global/page-templates/page-template-popup/index.js [hunk 2]
```diff
@@ -82,7 +83,7 @@ const PageTemplatePopup = ({
     kind: yup.string().equals([PAGES_MODULE_KINDS.MEDIA]),
     name: yup.string().required(T.translate("validation.required")),
     type: yup.string().required(T.translate("validation.required")),
-    upload_deadline: yup.date().required(T.translate("validation.required")),
+    upload_deadline: yup.date().nullable(),
```
Comment (smarcet) [inline src/pages/sponsors-global/page-templates/page-template-popup/index.js]: @priscila-moneo this change makes `upload_deadline` optional for **all** consumers of the component, not just the global context.

The ticket required:
- Global template -> hide the field + make it optional (correct)
- Summit/sponsor template -> keep the field visible **and still required** (broken by this change)

The other two consumers of `PageTemplatePopup` - `show-pages-list-page/index.js` and `sponsor-pages-tab/index.js` - do not pass `isGlobal`, still render the date picker, but now allow the form to submit without a value because the validation is `.nullable()` instead of `.required()`.

**Suggested fix:** make the schema conditional using Formik's `validationContext`:

```js
// In useFormik:
validationContext: { isGlobal }

// In mediaModuleSchema:
upload_deadline: yup.date().when('$isGlobal', {
  is: true,
  then: (s) => s.nullable(),
  otherwise: (s) => s.required(T.translate("validation.required"))
})
```

**Review by smarcet — CHANGES_REQUESTED** [review]
@priscila-moneo please review

File: src/pages/sponsors-global/page-templates/page-template-popup/index.js [hunk 3]
```diff
@@ -33,6 +33,74 @@ import {
...
+        .min(1, T.translate("validation.file_required")),
+    otherwise: (schema) => schema.nullable()
+  })
+});
+
+const mediaModuleSchema = yup.object().shape({
+  kind: yup.string().equals([PAGES_MODULE_KINDS.MEDIA]),
+  name: yup.string().required(T.translate("validation.required")),
+  type: yup.string().required(T.translate("validation.required")),
+  upload_deadline: yup.date().when("$isGlobal", {
```
Comment (smarcet) [inline src/pages/sponsors-global/page-templates/page-template-popup/index.js:70]: @tomrndom This is the crux of the fix - `upload_deadline` becomes nullable when `$isGlobal` is true, required otherwise - but nothing in the test suite exercises this branch. All the new/changed tests in `page-template.test.js` cover `normalizePageTemplateModules`, a downstream payload-cleanup helper that runs after validation already passed; none of them validate the schema itself or render `MediaRequestModule`/`PageModules` with `isGlobal={true}` to assert the datepicker is hidden. A regression here (e.g. a typo turning `"$isGlobal"` into `"isGlobal"`, or dropping `context: { isGlobal }` from the `validateSync` call below) would silently stop enforcing the required deadline for non-global templates and ship undetected.

Suggested fix: add a test asserting `validationSchema`/`mediaModuleSchema` rejects a missing `upload_deadline` when `isGlobal` is false and accepts it when true, and extend the existing `page-template-module-form.test.js` to assert the datepicker is absent when `isGlobal` is true.

File: src/pages/sponsors-global/page-templates/page-template-popup/modules/page-template-media-request-module.js [hunk 4]
```diff
@@ -57,15 +66,17 @@ const MediaRequestModule = ({ baseName, index, mediaFileTypes }) => {
...
-      <Grid2 size={6}>
-        <InputLabel htmlFor={buildFieldName("upload_deadline")}>
-          {T.translate("page_template_list.page_crud.upload_deadline")}
-        </InputLabel>
-        <MuiFormikDatepicker
-          name={buildFieldName("upload_deadline")}
-          margin="none"
-        />
-      </Grid2>
+      {showUploadDeadline && (
```
Comment (smarcet) [inline src/pages/sponsors-global/page-templates/page-template-popup/modules/page-template-media-request-module.js:77]: @tomrndom Hiding the datepicker here only removes it from the UI - it doesn't clear `upload_deadline` in Formik state. For an existing global page template whose MEDIA module already carries a legacy `upload_deadline` (e.g. data created before this fix), `denormalizePageModules` still hydrates that value into form state, the field stays truthy but invisible, and `normalizePageTemplateModules` (`src/utils/page-template.js:45-50`) only `delete`s the field when it's falsy — so the stale timestamp is silently re-sent on every save with no way for the admin to see or clear it.

Suggested fix: when `isGlobal` is true and the module kind is MEDIA, explicitly reset `upload_deadline` to `null` (e.g. via `setFieldValue` in `PageModules` or on mount in `MediaRequestModule`) so stale deadlines don't survive an edit-and-save cycle.

File: src/pages/sponsors-global/page-templates/page-template-popup/index.js [hunk 5]
```diff
@@ -53,82 +122,34 @@ const PageTemplatePopup = ({
...
-      code: yup.string().required(T.translate("validation.required")),
-      name: yup.string().required(T.translate("validation.required")),
-      ...(showSponsorships && {
-        sponsorship_types: yup
-          .array()
-          .min(1, T.translate("validation.required"))
-      }),
-      modules: yup.array().of(moduleSchema)
-    }),
+    validate: (values) => {
```
Comment (smarcet) [inline src/pages/sponsors-global/page-templates/page-template-popup/index.js:153]: @tomrndom Switching from the declarative `validationSchema` prop to this manual `validate` callback was necessary to pass `context: { isGlobal }` into Yup, but it also drops Formik's built-in `prepareDataForValidation` step (which normalizes `''` to `undefined` before validation runs) for every field in the form, not just `upload_deadline`. I checked this repo's yup (1.7.1): `.required()` still rejects `''` directly, so required-field enforcement isn't broken. But non-required numeric fields cleared to `''` (e.g. `max_file_size`) now surface as a Yup type-cast error ("must be a number type... cast from the value \"\"") instead of Formik's normalized "required" message - a behavior change to every field's validation UX as a side effect of this fix, and it isn't called out or tested.

Suggested fix: coerce `''` to `undefined` on `values` before calling `validationSchema.validateSync(...)`, mirroring Formik's `prepareDataForValidation`, so cleared fields keep producing the same "required" error message they did before this refactor.

File: src/pages/sponsors-global/page-templates/page-template-popup/modules/page-template-media-request-module.js [hunk 6]
```diff
@@ -1,23 +1,43 @@
...
     getIn(values, buildFieldName("type")) || PAGE_MODULES_MEDIA_TYPES.FILE;
 
+  // clears a stale upload_deadline hydrated from a legacy module once the field is hidden;
+  // only needs to run on mount, since showUploadDeadline doesn't change during a module's lifetime
+  useEffect(() => {
+    if (
+      !showUploadDeadline &&
+      getIn(values, buildFieldName("upload_deadline"))
+    ) {
+      setFieldValue(buildFieldName("upload_deadline"), null);
```
Comment (santipalenque) [inline src/pages/sponsors-global/page-templates/page-template-popup/modules/page-template-media-request-module.js:34]: can't we just set the upload_deadline to null when showUploadDeadline is false ? why do we need to check if it has a value ?

File: src/pages/sponsors-global/page-templates/page-template-popup/index.js (hunk 5 above)
Comment (santipalenque) [reply src/pages/sponsors-global/page-templates/page-template-popup/index.js:153]: @tomrndom  why don't we just add a hidden field with the isGlobal value ? then we can just have normal conditional validation . I think thats more straight forward

File: src/pages/sponsors-global/page-templates/page-template-list-page.js [hunk 7]
```diff
@@ -114,9 +114,17 @@ const PageTemplateListPage = ({
     setOpenCloneDialog(true);
   };
 
-  const handleSavePageTemplate = (entity) => {
-    savePageTemplate(entity).then(() => setOpenPageDialog(false));
-  };
+  const handleSavePageTemplate = (entity) =>
+    savePageTemplate(entity).then(() =>
```
Comment (smarcet) [inline src/pages/sponsors-global/page-templates/page-template-list-page.js]: @tomrndom This handler's own refetch races the refetch that `savePageTemplate` already fires internally, and this pattern was also introduced in commit 7a121bac.

`savePageTemplate` (`src/actions/page-template-actions.js`) dispatches `getPageTemplates()(dispatch)` with no arguments inside its own `.then()` — resetting term/page/order/archived-filter — fire-and-forget. This handler now fires a *second*, correctly-parameterized `getPageTemplates(term, DEFAULT_CURRENT_PAGE, perPage, order, orderDir, showArchived)` right after. Both requests hit the same list endpoint concurrently and dispatch into the same reducer slice; whichever response lands last wins. If you're searching, on page 2, or showing archived items and you edit+save a template, the visible list can silently snap back to page 1 / no filter / unarchived depending on network timing.

This also delays the popup's close: because this arrow function implicitly returns `getPageTemplates(...).catch(() => {})`, the promise `PageTemplatePopup` awaits before calling `onClose()` now waits on this second refetch too — unlike the sibling handlers added in this same PR (`handleSaveCustomizedPage`, `handleSaveManagedPage`, `handleSaveShowPage`), which correctly use a block body so the refresh is fire-and-forget, per the project's popup pattern (`summit-admin-popup-dialog-pattern.md`: "the returned promise must track ONLY the save — the list refresh is fire-and-forget").

Suggested fix: drop the internal blind `getPageTemplates()(dispatch)` from `savePageTemplate`'s two branches (or thread real filter args through it instead of calling with no args), and make this handler's refetch a fire-and-forget block body like its siblings, e.g. `savePageTemplate(entity).then(() => { getPageTemplates(term, DEFAULT_CURRENT_PAGE, perPage, order, orderDir, showArchived).catch(() => {}); })`.

File: src/pages/sponsors-global/page-templates/page-template-list-page.js [hunk 8]
```diff
@@ -114,9 +114,17 @@ const PageTemplateListPage = ({
     setOpenCloneDialog(true);
   };
 
-  const handleSavePageTemplate = (entity) => {
-    savePageTemplate(entity).then(() => setOpenPageDialog(false));
-  };
+  const handleSavePageTemplate = (entity) =>
```
Comment (smarcet) [inline src/pages/sponsors-global/page-templates/page-template-list-page.js:117]: @tomrndom The `<PageTemplatePopup>` rendered further down in this file (around `onSave={handleSavePageTemplate}`, right below this function) is missing `isGlobal`, so it always treats this page as non-global — the `upload_deadline` field stays visible and required here too, which defeats the fix this PR is meant to deliver. GitHub won't let me anchor a comment directly on that JSX block since it falls outside this PR's diff hunks (see why below), so I'm leaving this note on the nearest touched line instead.

This isn't a new gap: commit 738b65e0 ("fix: page templates date field required") added `isGlobal` right there, and it was still present after 49f0b0a0. It was silently dropped in 7a121bac ("fix: add missing catch for errors on handleSavePageTemplate") — the same commit that rewrote this function — whose diff only touches `handleSavePageTemplate` in its commit message/intent but also removes that line as a side effect:
```diff
           onSave={handleSavePageTemplate}
-          isGlobal
         />
```
Because it was added in an early commit and removed in the latest one, the squashed PR diff against `master` shows zero change at that line, so it doesn't surface in `gh pr diff`, CodeRabbit, or Copilot's review — this is why it slipped through despite several review passes.

Suggested fix: restore `isGlobal` on that `<PageTemplatePopup>` (this page only ever edits global templates, so it should always be `true`), and add a test that renders `PageTemplateListPage` and asserts the prop reaches `PageTemplatePopup` — none of the current tests exercise this page-to-popup wiring, only the popup in isolation.

**Review by smarcet — CHANGES_REQUESTED** [review]
@tomrndom please review comments

File: src/actions/page-template-actions.js [hunk 9]
```diff
@@ -175,10 +175,9 @@ export const savePageTemplate = (entity) => async (dispatch) => {
             html: T.translate("page_template_list.page_crud.page_saved")
           })
         );
-        getPageTemplates()(dispatch);
       })
       .catch((err) => {
-        console.error(err);
+        throw err;
```
Comment (santipalenque) [inline src/actions/page-template-actions.js:180]: you are already returning the promise so this throw is redundant

File: src/actions/page-template-actions.js [hunk 10]
```diff
@@ -200,10 +199,9 @@ export const savePageTemplate = (entity) => async (dispatch) => {
           html: T.translate("page_template_list.page_crud.page_created")
         })
       );
-      getPageTemplates()(dispatch);
     })
     .catch((err) => {
-      console.error(err);
+      throw err;
```
Comment (santipalenque) [inline src/actions/page-template-actions.js:204]: you are already returning the promise so this throw is redundant

**Review by santipalenque — APPROVED** [review]
LGTM

**Review by smarcet — APPROVED** [review]
LGTM
